src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1024,18 +1024,37 @@ def visit_Index(self, node: ast.Index):
         index = self.visit(node.value)
         return index
 
-    def _eval_index(self, node: ast.Subscript) -> Optional[List[int]]:
-        invalid_target = GTScriptSyntaxError(message="Invalid target in assignment.", loc=node)
-
+    def _eval_index(
+        self, node: ast.Subscript, axes: Optional[Set[str]] = None
```

Comment:
The current implementation seems to only support passing "I", "J" or "K", so I would check that `axes` only contain valid axis names (at least in typing)
```suggestion
        self, node: ast.Subscript, axes: Optional[Set[Literal["I", "J", "K"]]] = None
```
One enhancement we could consider though, is supporting external symbols in the offset, which would work as an alias to one of the cartesian axes:

```python
@stencil(... externals={"AXIS": gtscript.I})
def stencil(field):
    ....
    return field[AXIS - 1] + field[AXIS + 1]
```

Reply:
Thanks. I had this thought last night too. One of the key benefits of this is the ability to do that, as well as writing generic code that can work with 1d, 2d, or 3d fields, e.g. in functions.